### PR TITLE
#24237 add Hong Kong, Taiwan as Chinese states; fix name of Macao

### LINKED ISSUE #24237: Country issue
**Describe the bug**
A clear and concise description of what the bug is. Please be as descriptive as possible; issues lacking detail, or for any other reason than to report a bug, may be closed without action.

**To Reproduce**

**Screenshots**

**Expected behavior**

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin.
```
</details>

### PR DESCRIPTION
**Pull request by nobodxbodon**
### All Submissions:

* [ ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #24237. Actually it's already closed, but as mentioned in the [comment](https://github.com/woocommerce/woocommerce/issues/24237#issuecomment-516976282) there appears to be several items missing in the Chinese state list. Also there seems to be a typo in the spelling of Macao, according to [ISO 3166](https://www.iso.org/obp/ui/#iso:code:3166:MO).

### How to test the changes in this Pull Request:

1. 
2.
3.

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Seems like "Macau" is also accepted: https://en.wikipedia.org/wiki/Macau

But I want to revise the ISO codes, if there's 2 letters ISO, why is using those numbers in here? Do you have any source or how it works on Google?

**Comment by nobodxbodon** [comment]
Thanks @claudiosanches for reviewing.

> Seems like "Macau" is also accepted: https://en.wikipedia.org/wiki/Macau

The [header comment](https://github.com/woocommerce/woocommerce/blob/500373f34049d60fb5da7671094eaf276756ab28/i18n/states.php#L7) states:
> Country codes and states (or province) names should follow the ISO_3166-2 standard (https://en.wikipedia.org/wiki/ISO_3166-2).

According to [ISO 3166](https://www.iso.org/obp/ui/#iso:code:3166:MO) it's `Macao`. Besides it keeps consistent with [here](https://github.com/woocommerce/woocommerce/blob/500373f34049d60fb5da7671094eaf276756ab28/i18n/countries.php#L142).

Though these three have Alpha-2 codes in ISO 3166, their `independent` attributes are all `No`, as they are part of China. So I suppose Hong Kong and Taiwan shall be in the state list of China, similar to Macao?

**Comment by claudiosanches** [comment]
> According to ISO 3166 it's Macao. Besides it keeps consistent with here.

Now since it's part of Chine, we should remove it as a country.

> Though these three have Alpha-2 codes in ISO 3166, their independent attributes are all No, as they are part of China. So I suppose Hong Kong and Taiwan shall be in the state list of China, similar to Macao?

Sounds like you didn't understand. I want to use Alpha-2 codes for all entries related to China, since you already show to us that there's Alpha-2 codes for those states.

**Comment by nobodxbodon** [comment]
> I want to use Alpha-2 codes for all entries related to China, since you already show to us that there's Alpha-2 codes for those states.

OK may I ask what additional changes I need to make?

**Comment by nobodxbodon** [comment]
@claudiosanches may I ask if you meant using the [3166-2 code](https://www.iso.org/obp/ui/#iso:code:3166:CN) for the provinces? For example, `CN-MO` for Macao, instead of current `CN30`; `CN-AH` for Anhui instead of `CN13`?

**Comment by unfulvio** [comment]
I would suggest against this PR as ISO country lists are politically biased (that goes for any sensitive territory in the World, so setting this might be a precedent that opens a pandora box). 

Without entering into the details of the matter or the motivations one might have for or against, the UNICODE list provide a much more politically neutral approach to list countries in a more loose way. 

Please check this https://www.drupal.org/project/drupal/issues/1938892 issue for Drupal concerning the same matter where they bring more educated argumentation to the matter.

**Comment by ucheng** [comment]
Following ISO-3166 country list is a not a good decision and will make more confusion for people. It has a long historical debate, and can't reflect the current situation between Taiwan and China.

The country data in ISO-3166 country list originally came from UN.  Since Taiwan is not a member of UN, it's really hard to change the name. That's a political issue.

I do agree with @unfulvio on using the CLDR unicode data which is much more update-to-date.

Also, there is a relative issue in the upcoming WordCamp Asia about the country name of Taiwan.
https://meta.trac.wordpress.org/ticket/4638

**Comment by ucheng** [comment]
> @unfulvio as woocommerce is currently based on ISO 3166, IMO switching to another standard would be another topic. A new issue seems to be more proper, and doesn't need to block this PR.

If the ISO 3166 spec is not right and controversial, why should we follow it?

**Comment by yoren** [comment]
PRs are usually for bug fixes or enhancements which would benefit users once it got merged. As a long-time user of WooCommerce, I don't see any confusion right now that WooCommerce lists Taiwan as "Taiwan" in the country list. And I am also not sure who will be benefited after it gets changed to "Taiwan, Province of China" or listed as a state of China.

ISO 3166 isn't a golden rule that we, the developer community, should value over user experiences. From the links @unfulvio and @ucheng provided we can already see how some precedences were handled. So it would be great if they can also be considered here.

**Comment by yoren** [comment]
Hi @claudiosanches What we tried to convey here is to clarify ISO 3166 is historically not as neutral as you think and in other precedences (from Drupal https://www.drupal.org/project/drupal/issues/1938892 and WordCamp https://meta.trac.wordpress.org/ticket/4638) they already come to an agreement to adopt CLDR for such a reason, when handling the country names with disputes listed in ISO 3166 (https://en.wikipedia.org/wiki/ISO_3166-1#Naming_and_disputes).

> if this is not the case for China, please send us the full list how you use to do for any other eCommerce platform in China

In short, we, the WordPress and WooCommerce community in Taiwan, would not consider ourself to be included in the case here. Because in practices, while most China eCommerce platform may list Taiwan as its state, none of the eCommerce systems in Taiwan does so. We would like to stay as it is in WooCommerce as it never causes any confusion for us.

**Comment by huanyichuang** [comment]
I agree with @yoren how ISO 3166 is not as neutral as you think. For the current practices in Taiwan, not China, we don't list Taiwan as "Province of China." And neither do other international eCommerce platforms such as Amazon do that.

**Comment by nobodxbodon** [comment]
This PR is meant to get the country/states data more consistent with the latest ISO 3166 standard, which has been used as reference in the project.

As I suggested [above](https://github.com/woocommerce/woocommerce/pull/24425#issuecomment-525585229), switching to another standard for the whole project is out of the scope of this PR. 

I'll be happy to join this discussion in another issue, including studying the latest CLDR, while learning more about the woocommerce features to better understand how the country/states data will be presented in UI. Still, I don't think that should block this PR.

**Comment by yoren** [comment]
If we are already aware of CLDR as a new convention in the software developer community, and willing to open a new discussion on it, why is it so urgent to merge this PR in favor of ISO 3166 at this point?

An open discussion should last as long as the agreement emerged gradually. Please understand our intention is not to block your PR, @nobodxbodon. We speak up because we know how this could affect the WooCommerce community in Taiwan, and we'd be grateful if you and @claudiosanches can take it into consideration.

We'd be happy to keep the discussion going and for now, I don't think we can come to an agreement on how Taiwan was handled with this PR. Thanks.

**Comment by ucheng** [comment]
hi @nobodxbodon, not sure how will this PR benefit the WooCommerce project and the community, or you just want every open source project obeys the "One China Policy"?

https://github.com/nobodxbodon/one-China

As a citizen of the WordPress and WooCommerce community or even more, the open source community, we should embrace everyone's voice respectfully and I think we have the right to name our country.

**Comment by nobodxbodon** [comment]
I don't think any open-source project has to obey certain rules, or be impacted by certain group's pressure, because everyone has their own view.

On the other hand, commercial software needs to comply with regulations in the regions that they have business with. Different regions have different regulations for sure. That's why we need international standards.

If a project has already chosen to follow certain standard, ISO 3166 in this case, it's better to keep consistent with it as much as possible, to avoid influences of different personal opinions, which is the exact purpose of standards.

**Comment by claudiosanches** [comment]
@yoren sorry, but I'm getting upset with way you are talking to me, and is starting to offend me, as I said, I'm trying to understand how it works, there's no rush to merge this PR, I just asked for you all how works in China and Taiwan.

**Comment by claudiosanches** [comment]
So let's try it, and I hope we all are here trying to make the better for everyone, so with this in mind, let's try:

- Can we keep "Taiwan" as a country with no "(Province of China)" mention?
- Can we also keep this new list including Macao, Hong Kong and Taiwan as states of China? Based on the comments here this seems to probably be used only in China, and note still we'll keep Macao, Hong Kong and Taiwan in the list of countries.

You all think that this should help everyone?

**Comment by nobodxbodon** [comment]
@claudiosanches sounds good to me, but if we leave "(Province of China)" out, one more thing to confirm.

May I ask how's the country list currently presented in the UI? If the selector is labeled as "Country" only, it can cause controversy.

In case there are such labels, using "Country / Region" or some expression alike would be more proper, like [Apple.com](https://www.apple.com/choose-country-region/).

**Comment by jonathanbossenger** [comment]
@claudiosanches please excuse my crude attempt to simplify the concerns, but the crux of the problem you face with this PR is this:

> for sure on WooCommerce we prefer what is common in any single region.

If you were in China, the common name for Taiwan would probably be as this PR has suggested (or close to it) see this example from Alibaba

https://drive.google.com/file/d/1OeZ6ke8jWs-ITNp4wD3jwXKFJETYHXo7/view?usp=sharing

If you were in Taiwan (or based on the example below, the US), the common name for Taiwan would probably be just that, see this example from Amazon

https://drive.google.com/file/d/1vsPG0jDvL-PmQgIKamXQmt0YVKWQslhk/view?usp=sharing

However, as the PR submitter has correctly stated, WooCommerce has chosen to follow the ISO 3166 naming standard in the 8th of August 2019, which is when the header comment in the states.php file was added.

https://github.com/woocommerce/woocommerce/commit/e6394ace273377185fb6c6a489d7852327de7f58#diff-bc5a8264d55435c59c3034e2b0ce8105

One would need to research the history of the standard as it applies to Taiwan to understand why the people of Taiwan disagree with that standard. 

So in this specific case, your decision to accept the PR as it stands is either one of following the current standard to the letter, a decision that seems to have been made earlier this month, based on the header comment, or following what is commonly used in any single region. However, to determine the latter, will require looking at the country lists from more sample regions than the ones I've posted above. 

Alternatively, as the PR submitter and others have pointed out, perhaps this conversation leads to a more in depth conversation about the neutrality of the standard chosen, and another standard should be considered? I'm happy to assist with researching this option to see if this might be viable, and can be accepted by everyone.

**Comment by jonathanbossenger** [comment]
@nobodxbodon raises a possible compromise, instead of changing the country and states lists, rather change the naming convention of the UI. As pointed out in the [apple.com](https://www.apple.com/choose-country-region/) example China, Hong Kong and Taiwan are listed as separate under the Asia Pacific region.

**Comment by Lin-Buo-Ren** [comment]
@claudiosanches 
The ISO 3166 standard is in great influence of UN's opinions(especially, PRC's opinions), which is pretty well-known to be disputed <https://en.wikipedia.org/wiki/ISO_3166-1#Naming_and_disputes>.  Here are some references for other projects:

* [Bug #614236 “Wrong Country Name. (Taiwan)” : Bugs : LoCo Team Portal](https://bugs.launchpad.net/loco-team-portal/+bug/614236)  
  (patched to use Debian-specific common_name field when available, which is "Taiwan")
* (patched to use Debian's common_name "Taiwan" when available)[Bug #1138121 “Taiwan is PRACTICALLY a “<country>”, but not “Prov...” : Bugs : software-properties package : Ubuntu](https://bugs.launchpad.net/ubuntu/+source/software-properties/+bug/1138121)  
  (patched to use Debian-specific common_name field when available, which is "Taiwan")
* (patched to use Debian's common_name when available)[Politically incorrect region name "Taiwan, Province of China" · Issue #310 · canonical-web-and-design/snapcraft.io](https://github.com/canonical-web-and-design/snapcraft.io/issues/310)  
  (patched to use Debian-specific common_name field when available, which is "Taiwan")
* [FS#30444 : Download Page uses Taiwan's name as “Taiwan, Province of China”](https://bugs.archlinux.org/task/30444)  
  (somehow fixed even though the bug is closed)

As a Taiwanese from a disputed sovereign state I'd like to note that straight out changing the names without other UX mitigations (like alternative name variants for different audience/government policies (*cough* One-China Policy*cough*)) is a no-go as it definitely offend lots of people.

I'd rather encourage a product:

* Avoid strictly following a "country name" standard as it is, in its nature, a political minefield.  For projects that uses [Debian's iso-codes package](https://salsa.debian.org/debian/iso-codes), the `common_name` field is a slightly better (but not perfect) source of neutral state-like names.
* Prefer "Regions" instead of "Countries" in the region selection UI
* Provide a "Make China-nese people happy" opt-in switch that use the OCP-compliant(to be honest here, it's the only "standard" what @nobodxbodon would really care, as stated in <https://github.com/nobodxbodon/one-China>) names if a developer prefers its representation.

**Comment by Lin-Buo-Ren** [comment]
@nobodxbodon 
Hello.  I would like to encourage you to split the Macao state name typo fix to a separate PR as it is a different category of issue.

**Comment by yoren** [comment]
In response to some of the latest comments, from my point of view:

1. To me and WC users in Taiwan, we'd like to stay as it is, which is: 1) list as "Taiwan" in the Country list; and 2) Do not be included in the States list of China.
2. If it must be changed, the common practices we can see and have been shown above, in other open-source projects or big eCommerce platforms like Amazon, the UI is named "Country/Region" where leaves "Taiwan" as simply "Taiwan".
3. If adopting this UI re-name approach as #2, Amazon doesn't have a state dropdown for China so I don't have an example. As for WooCommerce, I don't agree Taiwan should be listed as a Chinese state. Besides we don't agree with ISO 3166, please also note technically no one will select "China" and then pick "Taiwan" as the state in their shipping or billing address if they live in Taiwan. I doubt if they can ever get any delivery to such address. So UX is still the main concern here.

Lastly, if what I stated or the way I talked in this thread offend anyone, please understand that was definitely not my intention. Taiwan is a beautiful country and always embraces different voices and we encourage people to speak up for themselves. I'm just trying to participate in the discussion with the same spirit. Thanks for reading this far.

**Comment by justin-lau** [comment]
> In response to some of the latest comments, from my point of view:
> 
> 1. To me and WC users in Taiwan, we'd like to stay as it is, which is: 1) list as "Taiwan" in the Country list; and 2) Do not be included in the States list of China.
> 2. If it must be changed, the common practices we can see and have been shown above, in other open-source projects or big eCommerce platforms like Amazon, the UI is named "Country/Region" where leaves "Taiwan" as simply "Taiwan".
> 3. If adopting this UI re-name approach, Amazon doesn't have a state dropdown for China so I don't have an example. As for WooCommerce, I don't agree Taiwan should be listed as a Chinese state. Besides we don't agree with ISO 3166, please also note technically no one will select "China" and then pick "Taiwan" as the state in their shipping or billing address if they live in Taiwan. I doubt if they can ever get any delivery to such address. So UX is still the main concern here.
> 
> Lastly, if what I stated or the way I talked in this thread offend anyone, please understand that was definitely not my intention. Taiwan is a beautiful country and always embraces different voices and we encourage people to speak up for themselves. I'm just trying to participate in the discussion with the same spirit. Thanks for reading this far.

Same comment from a HongKonger. No one who lives in Hong Kong would pick China as the country before looking up Hong Kong from the list of provinces in any address form.

For historical reason, people are already used to picking Hong Kong from a country list. If they are not listed we will more likely assume these location are simply unavailable.

**Comment by claudiosanches** [comment]
And seems like I just got completed ignored in this PR again!
I see that it's possible to have a nice conversation around this subject, people trying to offend me, they don't even care to say "sorry", there's no collaboration to find a solution.
So I'm closing it, this already took to many time, and there's filters if you'll prefer to change it.

**Comment by claudiosanches** [comment]
Despite the use of ISO 3166 in WooCommerce, we have always tried to include everyone, so I'm officially saying that WooCommerce adopts or tries to adopt CLDR, changes in this regard should be made when necessary and we hope to count on the help and understanding of the community.

So we are going to revert this change https://github.com/woocommerce/woocommerce/commit/e6394ace273377185fb6c6a489d7852327de7f58 since isn't like we really does on WooCommerce, and going to update with CLDR docs.

Also if CLDR isn't a solution for you, please feel free to use filters like `woocommerce_countries`, `woocommerce_states`, `woocommerce_continents` and more that can be used to build custom plugins to attend determinate needs, take for example some places like Germany and Brazil that have plugins to make WooCommerce work better in those places, including new fields to collection IDs and any data necessary, or even making changes to help the life of users in those locations.

Thank you everyone for join this conversation.

**Comment by rodrigoprimo** [comment]
Thanks, everyone for your input. I can say that I learned a lot from your comments and from reading more about CLDR. As Claudio mentioned above, we (the WooCommerce core team) had a conversation about this topic, and we believe that the best course of action is to use CLDR instead of ISO 3166. See #24126. 

> However, as the PR submitter has correctly stated, WooCommerce has chosen to follow the ISO 3166 naming standard in the 8th of August 2019, which is when the header comment in the states.php file was added.

Sorry for the delay to answer and it doesn't matter much now that we will stop using ISO 3166, but just to clarify that it is not that we chose to follow ISO 3166 last month. WooCommerce has been using this standard for a long time, commit e6394ac simply made that more explicitly by documenting this in the states.php file.

**Comment by rodrigoprimo** [comment]
I just realized (thanks, @peterfabian) that initially I linked to the wrong PR in the comment above. The correct PR is #24564 

I updated the original comment.
